llvm/test/CodeGen/AMDGPU/GlobalISel/assert-align.ll

```diff
@@ -44,7 +46,6 @@ entry:
 define ptr addrspace(1) @tail_call_assert_align() {
 ; CHECK-LABEL: tail_call_assert_align:
 ; CHECK:       ; %bb.0: ; %entry
-; CHECK-NEXT:    s_waitcnt vmcnt(0) expcnt(0) lgkmcnt(0)
```

Comment:
Why do these drop completely here, but in the next test (`br-constant-invalid-sgpr-copy.ll`) they just move to the end of the function? Is it because this function returns something, so it's the caller's job to wait before reading the results?

Reply:
This function ends with a tail call, not a return. The ABI says that counters are undefined on function entry but must be 0 on return.